sdk/keyvault/azkeys/constants.go

```diff
@@ -133,16 +169,77 @@ func (c ActionType) ToPtr() *ActionType {
 	return &c
 }
 
-// KeyEncryptionAlgorithm - The encryption algorithm to use to protected the exported key material
-type KeyEncryptionAlgorithm string
+// PossibleActionTypeValues provides a slice of all possible ActionTypes
+func PossibleActionTypeValues() []ActionType {
+	return []ActionType{
+		ActionTypeRotate,
+		ActionTypeNotify,
+	}
+}
+
+// EncryptionAlgorithm - The encryption algorithm to use to protected the exported key material
+type EncryptionAlgorithm string
 
 const (
-	KeyEncryptionAlgorithmCKMRSAAESKEYWRAP KeyEncryptionAlgorithm = "CKM_RSA_AES_KEY_WRAP"
-	KeyEncryptionAlgorithmRSAAESKEYWRAP256 KeyEncryptionAlgorithm = "RSA_AES_KEY_WRAP_256"
-	KeyEncryptionAlgorithmRSAAESKEYWRAP384 KeyEncryptionAlgorithm = "RSA_AES_KEY_WRAP_384"
+	EncryptionAlgorithmCKMRSAAESKEYWRAP EncryptionAlgorithm = "CKM_RSA_AES_KEY_WRAP"
```

Comment:
Yikes, the acronyms make this unreadable but I suppose doing anything to break them up is unconventional 🙄 

Should this be `KeyWrapAlgorithmCKMRSAAES`?

Reply:
Agree, they are ugly but I can't think of a better way to handle them.

Reply:
About the prefix then--will Key Vault allow these algorithms in `encrypt`?

Reply:
Should I remove the `KEYWRAP` portion from all the models in this scenario?

Reply:
Although "key wrap" is part of the algorithm name I think a `KeyWrap` prefix implies it strongly enough so, `KeyWrapAlgorithmRSAAES256` > `KeyWrapAlgorithmRSAAESKEYWRAP256`. @heaths, what do you think?

Reply:
I don't think "key wrap" should be removed from the algorithm name. They are long, but many, many constants in crypto are. Also, I don't think we should preface the constants with "KeyWrap" when used for the `Encrypt` function. The algorithms that `Encrypt` and `WrapKey` support are different.

Reply:
> The algorithms that `Encrypt` and `WrapKey` support are different.

That's what I was getting at originally. These constants are for `WrapKey`, so we should have something like
```suggestion
	KeyWrapAlgorithmCKMRSAAESKEYWRAP KeyWrapAlgorithm = "CKM_RSA_AES_KEY_WRAP"
```
But taking a closer look, I see we have another bunch of EncryptionAlgorithms and WrapAlgorithms in `crypto/constants.go`. What are these for?

Reply:
* `WrapAlgorithm` is used in the `crypto.Client.WrapKey/UnwrapKey`
* `EncryptionAlgorithm` is used in `crypto.Client.Encrypt/Decrypt`

Reply:
Ah, these constants I'm commenting on are for `ReleaseKey` only? .NET calls the type [KeyExportEncryptionAlgorithm](https://github.com/Azure/azure-sdk-for-net/blob/3b295806bf2bf3af6ec2380173363db131f931ca/sdk/keyvault/Azure.Security.KeyVault.Keys/src/ReleaseKeyOptions.cs#L33)

Reply:
In general, I would keep separate sets of constants for different methods (or sets of related methods) / models based on the swagger. That they are in different models means they can - and most likely will - diverge and that would make matters even more confusing for developers.

Reply:
Thanks for the catch @chlowell , I've updated the name